### ISSUE TITLE
Boxfile feature parity with Pagodabox Boxfile

### ISSUE THREAD

**Issue by frankbardon** [nanobox-io/nanobox#147]
I'm not quite certain if this is something that can be dealt with here, but I figured I'd ask. I'm working on setting up my team's local development environment using Nanobox. I'm finding that there are inconsistencies between the Boxfile I use for production and the one I'd use in the local environment. More specifically:

The naming convention `database1` on PB is not the same as the specific naming requirements of Nanobox `mariadb1`. The "type" parameter is not honored in Nanobox.

The `stability` parameter of each individual service does not match between environments. On PB, the parameter must be set to `production` where on NB it must be set to `stable`.

These are minor inconsistencies, true... but it prevents the Nanobox build from being able to complete without generating a new Boxfile for each environment. This can be accommodated with Git, but it requires managing a separate branch just to host a separate Boxfile.

I guess, the first question would have to be; is it your intention to maintain portability of the Boxfile between these two services? Judging by the description laid out in the blog post announcing Nanobox, it would seem that this was a primary concern. My $0.02 would have to be that it is important for Pagodabox customers to have the ability to test locally, and may be something that is taken into consideration.

The second question then becomes; have you run into this issue before, and what is the best way to deal with it?

**Comment by sanderson** [comment]
Hi Frank,

Nanobox is very similar to Pagoda Box, but very different in important ways. You probably already know this, but we are the same team behind both Pagoda Box and Nanobox. Nanobox is an iteration of Pagoda Box. Unfortunately, parity between the two projects is no longer a goal of the Nanobox project. In fact, major changes for the Nanobox Boxfile are in the works that will push it further away from Pagoda Box. More info is coming soon.

It's true that Nanobox originally started out as a local development tool for Pagoda Box, but we learned early on that we needed to diverge. Nanobox is on the verge of becoming its own separate service that will provide a seamless workflow from local development into production apps on your cloud provider of choice. If you're interested, you can sign up for the private beta on [nanobox.io](https://nanobox.io/).

**Comment by frankbardon** [comment]
Thank you for the explanation, I appreciate that very much. 

Would it be sensible to submit an issue allowing the possibility to rename the `Boxfile` on nanobox so that I do not have to maintain two when using the services together?

I think I understand the direction of Nanobox a little better now, and I'm excited about the possibilities. Just hoping that, in the meantime, I have a workable solution for our new development team. I recently convinced my boss to utilize Pagodabox instead of AWS so that we did not have to constantly fiddle with server maintenance and I've been pleased so far. However, it would be extremely nice to be able to test the entire stack locally.

Thank you again!
